Mathlib/Algebra/BigOperators/Pi.lean

```diff
@@ -57,6 +58,26 @@ theorem pi_eq_sum_univ {ι : Type*} [Fintype ι] [DecidableEq ι] {R : Type*} [S
   ext
   simp
 
+section CommSemiring
+variable [CommSemiring α]
+
+lemma prod_indicator_apply (s : Finset ι) (f : ι → Set κ) (g : ι → κ → α) (j : κ) :
+    ∏ i ∈ s, (f i).indicator (g i) j = (s.inf f).indicator (∏ i ∈ s, g i) j := by
+  rw [Set.indicator]
+  split_ifs with hj
+  · rw [Finset.prod_apply]
+    congr! 1 with i hi
+    simp only [Finset.inf_set_eq_iInter, Set.mem_iInter] at hj
+    exact Set.indicator_of_mem (hj _ hi) _
+  · obtain ⟨i, hi, hj⟩ := by simpa using hj
+    exact Finset.prod_eq_zero hi <| Set.indicator_of_not_mem hj _
+
+lemma prod_indicator (s : Finset ι) (f : ι → Set κ) (g : ι → κ → α) :
```

Comment:
Can you also add the version of this where `f` is constant (for rewrite chains, so the rhs is nice)

Reply:
I think you did the `g` constant one instead... Oh well

Reply:
Oh sorry! But the constant `f` version already has other lemmas for it

Reply:
Ah, where are they?

Reply:
Actually looks like we have `Finset.indicator_sum` and `Finset.mulIndicator_prod`, but not `Finset.indicator_prod`.

Reply:
Exactly, which is why I asked for this one :)

Reply:
In the future, please note that you will get better feedback to your comments if the commented lines actually comprise the statement. Here I saw you asked for constant `f`, but didn't immediately see what `f` was in the statement, I saw that `g` was a function and made that constant instead of `f`.

Reply:
Can you elaborate on this? The commented line does contain the statement. It includes `f`, in particular it includes the `f` which I wanted you to make constant. Is there a reason that `g` is more noticeable here than `f`?

Reply:
I mean, my advice above is a general advice. But here specifically I thought "Ah! Bhavik wants me to make the function constant" and I didn't think of `f` as the function (because it's set-valued) but `g` instead (because it's function-valued)

Reply:
I'm still not quite sure I follow. The line I highlighted was a deliberate choice, because it's where `f` is declared. The function `f` is a set-valued function, which is the one I wanted you to make constant. What would you have preferred here? Is there a way you'd prefer me to refer to a function other than by its name and pointing to the place where it's declared?

Reply:
Yes, you can comment on all the lines of the statement. Here L. 75-76. Multiline comments have been available on github for the past 2-3 years or so

Reply:
Just to clarify, your advice here is that it's less ambiguous to refer to changing the type of a function if I additionally highlight things which are not the type of the function? And that referring to an input `f` by its name and declaration line is not sufficiently clear, and you'd prefer seeing things other than inputs in a suggestion to amend an input? 
Can you clarify why you'd prefer this? And could you explain where the ambiguity arises, for my future reference? If there is a way to refer to a variable in a lemma by something other than its name and type, I would be very happy to hear it.
I appreciate the information about multiline comments on github, indeed I have been using them for the last 4-5 years or so.